src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+    if (nextIntersectionIsTooFarAway(intersection_node_id, next_road.eid))
     {
         return intersection;
     }
-    auto next_intersection_node = node_based_graph.GetTarget(next_road.eid);
 
-    const auto next_road_next_intersection =
-        findNextIntersectionForRoad(intersection_node_id, next_road, next_intersection_node);
+    // Try to find the intersection at (c) which the Sliproad shortcuts
+    const auto next = getNextIntersection(intersection_node_id, next_road);
 
-    if (next_road_next_intersection.empty())
+    if (!next)
         return intersection;
 
     // If we are at a traffic loop at the end of a road, don't consider it a sliproad
-    if (intersection_node_id == next_intersection_node)
+    if (intersection_node_id == next->node)
         return intersection;
 
-    std::unordered_set<NameID> target_road_names;
+    std::vector<NameID> target_road_name_ids;
+    target_road_name_ids.reserve(next->intersection.size());
 
-    for (const auto &road : next_road_next_intersection)
+    for (const auto &road : next->intersection)
     {
         const auto &target_data = node_based_graph.GetEdgeData(road.eid);
-        target_road_names.insert(target_data.name_id);
+        target_road_name_ids.push_back(target_data.name_id);
     }
 
-    for (auto &road : intersection)
+    // TODO: remove geojson debugging
+    NodeID b, c, d;
+    b = intersection_node_id;
+    c = next->node;
+
+    auto sliproad_found = false;
+
+    // Check all roads for Sliproads and assign appropriate TurnType
+    for (std::size_t road_index = 0, last = intersection.size(); road_index < last; ++road_index)
     {
-        if (linkTest(road))
+        auto &sliproad = intersection[road_index]; // this is what we're checking and assigning to
+
+        const auto &sliproad_data = node_based_graph.GetEdgeData(sliproad.eid);
+
+        // Discard service and other low priority roads - never Sliproad candidate
+        if (sliproad_data.road_classification.IsLowPriorityRoadClass())
+            continue;
+
+        // This is what we know so far:
+        //
+        //       .
+        //       e
+        //       .
+        //       .
+        // a ... b .... c .   < `next` is intersection at `c`
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d     < `target_intersection` is intersection at `d`
+        //              .       `sliproad_edge_target` is node `d`
+        //              e
+        //
+        //
+        //          ^ `sliproad` is `bd`
+        //       ^ `intersection` is intersection at `b`
+
+        if (!is_potential_link(sliproad))
+            continue;
+
+        // b -> d edge id
+        EdgeID sliproad_edge = sliproad.eid;
+
+        // TODO: remove debugging
+        d = node_based_graph.GetTarget(sliproad_edge);
+
+        const auto target_intersection = [&](NodeID node) {
+            auto intersection = intersection_generator(node, sliproad_edge);
```

Comment:
don't you have logic for skipping over these? Also: there is a high potential for multiple traffic lights. The intersection_generator as well offers `GetActualNextIntersection` (could use a better name) which does exactly what you do here, a bit more efficiently. Here you are generating coordinates for all traffic lights, but you don't care about these.